org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/DecompilerManager.java

```diff
@@ -0,0 +1,186 @@
+/*******************************************************************************
+ * Copyright (c) 2017 David Gileadi and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     David Gileadi - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.managers;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.ISafeRunnable;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.SafeRunner;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.IClassFile;
+import org.eclipse.jdt.ls.core.internal.Disassembler;
+import org.eclipse.jdt.ls.core.internal.IDecompiler;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.preferences.PreferenceManager;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+
+public class DecompilerManager {
+
+	public static final String DECOMPILED_HEADER = " // Failed to get sources. Instead, stub sources have been generated by the %s.";
+	public static final String DISASSEMBLER_HEADER_ADDITION = " // Implementation of methods is unavailable.\n";

```

Comment:
this comment belongs to the disassembler

Reply:
I waffled between the different ways and had decided that the job of adding messages belonged to DecompilerManager. However I'm plenty happy to switch it.